### PR TITLE
Add images of wordcloud for faster loading

### LINKED ISSUE #19: Save finalized state word clouds to images to speed up streamlit app load time

### PR DESCRIPTION
**Pull request by jkosh44**
Closes #19

### CODE REVIEW

**Comment by jkosh44** [comment]
The word clouds should also be a lot more consistent in terms of size and quality, so just checkout this branch and test out a couple of states. Alabama was one of the issue states which looks a lot better in this PR.

**Comment by jkosh44** [comment]
Also the word cloud sizes and bar chart x axis length are easily changed, so let me know thoughts on that.

File: streamlit_app.py [hunk 1]
```diff
@@ -42,13 +54,18 @@ def get_word_df(words):
...
+        cached_pic = get_wordcloud_from_file(state)
+        if cached_pic:
+            st.image(cached_pic, width=600)
+        else:
+            cleaned_tweets = get_cleaned_tweet_words(state, stopwords)
+            wordcloud = get_wordcloud(cleaned_tweets, state)
+            st.pyplot(wordcloud)
     else:
+        cleaned_tweets = get_cleaned_tweet_words(state, stopwords)
         title = f"{state} Tweets" if state else "Global Tweets"
```
Comment (lee-vivian) [inline streamlit_app.py:67]: Not sure about this question but wanted to double-check, are these really global tweets or just US tweets?

**Review by lee-vivian — APPROVED** [review]

**Comment by lee-vivian** [comment]
I think the bar chart sizes look good and it definitely loads faster when I run it locally! Is there a way to also set a max height to the word clouds? Some state word clouds are still noticeably larger than others (but if this is limited by their inherent shapes, I think it's also okay to leave it as is). Overall looks great though!

File: streamlit_app.py (hunk 1 above)
Comment (jkosh44) [reply streamlit_app.py:67]: The site doesn't actually specify where the tweets are from but they never mention limiting them to one country: https://dataverse.harvard.edu/dataset.xhtml?persistentId=doi:10.7910/DVN/LW0BTB&version=7.0. Also when I was fetching and filtering them I was seeing tweets from other countries.

Comment (lee-vivian) [reply streamlit_app.py:67]: Ah cool, thanks for the clarification!

**Comment by jkosh44** [comment]
I've noticed that too and I think it has to do with the underlying image. I've tried to explicitly set the height with the Wordcloud API but it must ignore that and just use the underlying picture dimensions. 
`word_cloud = wordcloud.WordCloud(background_color="white", mask=state_mask, contour_width=2,
                                         contour_color="steelblue", height=400, width=800).generate(tweet_str)`
The `st.image` function only takes a width parameter and scales the height accordingly. 

The other option I could think of would be to scale the numpy array created from loading in the image with PIL. I can check if PIL has some built in functionality otherwise it might be pretty complicated.

**Comment by lee-vivian** [comment]
Okay sounds good! Definitely no worries if not - don't want to overcomplicate this since it's very minor and pretty low prio

**Comment by jkosh44** [comment]
I'll merge this and work on the height issue separately

**Pull request merged**